core/rule_generator.py

```diff
@@ -816,18 +817,42 @@ def tables(pattern_json: str, rewrite_json: str) -> list:
         patternTables = RuleGenerator.tablesOfASTJson(patternASTJson, [])
         rewriteTables = RuleGenerator.tablesOfASTJson(rewriteASTJson, [])
 
-        # TODO - patternTables should be superset of rewriteTables
-        #
+        # patternTables should be superset of rewriteTables
+        patternSet = defaultdict(list)
+        rewriteSet = defaultdict(list)
+
+        for table in patternTables:
+            if type(table['value']) is str and type(table['name']) is str:
+                patternSet[table['value']].append(table['name'])
+
+        for table in rewriteTables:
+            if type(table['value']) is str and type(table['name']) is str:
+                rewriteSet[table['value']].append(table['name'])
+
+        superSet = []
+        for patternValue, patternNames in patternSet.items():
+            rewriteNames = rewriteSet.get(patternValue, [])
+            # special case: 
+            #   if patternTable ONLY have {'value': 'employee', 'name': 'employee'}
+            #   and rewriteTable ONLY have {'value': 'employee', 'name': 'e1'}, 
+            #   we replace 'employee' with 'e1' as table alias
```

Comment:
```suggestion
            #   we replace 'employee' with 'e1' as table alias
            #   the purpose is for the next step when we replace tables with variables, 
            #   we should be able to know the patternTable and rewriteTable should be replaced with the same variable.
            #   This logic will much simpler once we complete refactoring the code to introduce an internal tree structure
            #   to represent the AST instead of the current JSON structure, where can easily determine if two tables
            #   are the same table with different alias names.
```